server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -42,6 +42,13 @@ object LivyConf {
   val LIVY_SPARK_MASTER = Entry("livy.spark.master", "local")
   val LIVY_SPARK_DEPLOY_MODE = Entry("livy.spark.deployMode", null)
 
+  // Two configurations to specify Spark and related Scala version. These are internal
+  // configurations will be set by LivyServer and used in session creation. It is not required to
+  // set usually unless running with unofficial Spark + Scala versions
+  // (like Spark 2.0 + Scala 2.10, Spark 1.6 + Scala 2.11)
+  val LIVY_SPARK_SCALA_VERSION = Entry("livy.spark.scalaVersion", null)
+  val LIVY_SPARK_VERSION = Entry("livy.spark.version", null)
```

Comment:
Ideally LivyConf should remain to be an immutable conf object. Also I think we are passing a lot of other objects (like SessionStore, LivyConf) from LivyServer to SessionServlets. We can extract them to a class plus the Scala and Spark version and pass them from LivyServer to Servlets.

Since we need to move forward with the Spark 2.0 IT change, let's deal with this later.